app/models/job_proxy_dispatcher.rb

```diff
@@ -90,6 +93,53 @@ def dispatch
     _log.info "Complete - Timings: #{t.inspect}"
   end
 
+  def dispatch_container_scan_jobs
+    ret, = Benchmark.realtime_block(:pending_container_jobs) { pending_container_jobs }
+    jobs_by_ems, images_to_ems = ret
+    Benchmark.current_realtime[:container_jobs_to_dispatch_count] = jobs_by_ems.values.reduce(0) { |m, o| m + o.length }
+    concurrent_ems_limit = Settings[:container_scanning][:concurrent_per_ems].to_i
+    jobs_by_ems.each do |ems_id, jobs|
+      dispatch_to_ems(ems_id, jobs, concurrent_ems_limit, images_to_ems)
+    end
+  end
+
+  def dispatch_to_ems(ems_id, jobs, concurrent_ems_limit, images_to_ems)
+    jobs.each do |job|
+      active_ems_scans = active_container_scans_by_zone_and_ems(images_to_ems)[@zone][ems_id]
+      if concurrent_ems_limit > 0 && active_ems_scans >= concurrent_ems_limit
+        _log.warn(
+          "SKIPPING remaining %s scan jobs for %s [%d] in dispatch since there are [%d] active scans in zone [%s]" %
+            [
+              ui_lookup(:table => ContainerImage.name),
+              ui_lookup(:table => ExtManagementSystem.name),
+              ems_id,
+              active_ems_scans,
+              @zone
+            ]
+        )
+        break
+      end
+      do_dispatch(job, ems_id)
```

Comment:
Please add a test for this, maybe you'll have to extract methods to test some of this logic.